libraries/AP_Airspeed/AP_Airspeed_Health.cpp

```diff
@@ -0,0 +1,75 @@
+#include <AP_Common/AP_Common.h>
+#include <GCS_MAVLink/GCS.h>
+#include <AP_Math/AP_Math.h>
+#include "AP_Airspeed.h"
+
+bool AP_Airspeed::check_airspeed_health(uint8_t i)
+{
+    bool airspeed_status=true;
+
+    if((AP_AIRSPEED_FAILURE_MASK_BASIC & _options) && airspeed_status && (AP::ahrs().get_max_wind() > 0)) {
+        airspeed_status = check_airspeed_health_basic(i);
+    }
+
+    return airspeed_status;
+}
+
+bool AP_Airspeed::check_airspeed_health_basic(uint8_t i)
+{
+    float aspeed = get_airspeed();
+    bool airspeed_status=true;
+
+
+    float wind_max = AP::ahrs().get_max_wind();
+    uint32_t current_time = AP_HAL::millis();
+    uint32_t time_since_last_call =  current_time - state[i].airspeed_failure_last_check;
+    if(time_since_last_call > AP_AIRSPEED_FAILURE_TIME_PERIOD_MS) {
```

Comment:
time_since_last_call doesn't need it's own variable. It's used twice. Just use:
` (current_time - state[i].airspeed_failure_last_check > AP_AIRSPEED_FAILURE_TIME_PERIOD_MS)`
like everywhere else in the codebase. Same goes for the one below. Also, if these are all in millisec, then please name then with _ms like all the other millisecond variables in the codebase
rename current_time to now_ms and state[i].airspeed_failure_last_check to state[i].failure.last_check_ms